sys-fs/reiser4progs/files/reiser4progs-2.0.5-clang.patch

```diff
@@ -0,0 +1,62 @@
+Author: Denis Pronin <dannftk@yandex.ru>
+
+https://bugs.gentoo.org/933169
+
+--- a/plugin/vol/volume.c
++++ b/plugin/vol/volume.c
+@@ -34,8 +34,8 @@
+ 	return max(min(val, high),low);
+ }
+ 
+-static int advise_stripe_size_simple(uint64_t *result, uint32_t block_size,
+-				     uint64_t block_count, int is_default,
++static int advise_stripe_size_simple(uint64_t *result, uint64_t block_size,
++				     uint32_t block_count, int is_default,
```

Comment:
In include/reiser4/plugin.h, we have this struct declaration:
```
/* Volume plugin */
typedef struct reiser4_vol_plug {
        reiser4_plug_t p;
        int (*advise_stripe_size)(uint64_t *result, uint64_t block_count,
                                  uint32_t block_size, int is_default,
                                  int forced);
        int (*advise_nr_segments)(uint64_t *result, int forced);
        int (*check_data_capacity)(uint64_t result, uint64_t block_count,
                                    int forced);
  uint64_t (*default_data_capacity)(uint64_t free_blocks, int is_data_brick);
} reiser4_vol_plug_t;

```
Note that the `uint64_t block_count` parameter comes *before* the `uint32_t block_size` parameter, which is opposite to the order in which they are declared for advise_stripe_size_simple.

I think the proper solution would be to swap the order of the block_count and block_size parameters in the function definitions in plugin/vol/volume.c.

Reply:
Looking at how the plugin code is actually invoked via progs/mkfs/mkfs.c, it looks like the order of the parameters in include/reiser4/plugin.h might be reversed. So the solution might be to reverse them there instead.

In any case, this deserves a more careful look.